extensions/Lily/SoundExpanded.js

```diff
@@ -0,0 +1,365 @@
+(function (Scratch) {
+  "use strict";
+
+  const vm = Scratch.vm;
+  const runtime = vm.runtime;
+  const soundCategory = runtime.ext_scratch3_sound;
+
+  /**
+   * @param {VM.BlockUtility} util
+   * @param {unknown} targetName
+   */
+  const getSpriteTargetByName = (util, targetName) => {
+    const nameString = Scratch.Cast.toString(targetName);
+    const target = util.target;
+    if (target.getName() === nameString) {
+      return target;
+    }
+    return util.runtime.getSpriteTargetByName(nameString);
+  };
+
+  class SoundExpanded {
+    getInfo() {
+      return {
+        id: "lmsSoundExpanded",
+        color1: "#CF63CF",
+        color2: "#C94FC9",
+        color3: "#BD42BD",
+        name: "Sound Expanded",
+        blocks: [
+          {
+            opcode: "startLooping",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "start looping [SOUND]",
+            arguments: {
+              SOUND: {
+                type: Scratch.ArgumentType.SOUND,
+              },
+              START: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 0,
+              },
+            },
+          },
+          {
+            opcode: "stopLooping",

```

Comment:
is it intended that this doesn't actually stop the sound immediately?

Reply:
yes, that's why there's a "stop sound" block.